### ISSUE TITLE
Publishing separate SPM Packages through Github's API

### ISSUE THREAD

**Issue by noppoMan** [swift-aws/aws-sdk-swift#80]
@jonnymacs @Yasumoto 
Now I'm thinking how we can automate to publish separate SPM Packages(eg. swift-aws/ec2) flow.

My plan is [Publishing separate SPM Packages through Github's API](https://developer.github.com/v3/git/commits/#create-a-commit) as described in title.

The work flow is...

1. Updating models and  running CodeGenerator
2. Make a PR
3. Merging
4. Make a Tag(must follow semantic versioning) and push it to github, Then the script for publishing packages(commit, push, tagging) is executed on Travis(if the tag is followed semantic versioning)

How do you think this?

**Comment by Yasumoto** [comment]
I think this is a pretty great idea. 👏 

An alternative I don't think will work (but I would like to try in parallel just in case) is to keep all of these together, but make a new `Target` for each service. If we were to still include a "global" `AWSSDKSwift` target then that'd still take "forever" to build, but I'm still curious to see the data.

**Comment by jonnymacs** [comment]
I thought about the target approach as well. But then you have to d/l all the other services that you don’t ultimately need. Sure they don’t compile but still a bit of a downer.

The way spm is designed I think we really need to just have a ton of git repos.

Short answer I think Yuki’s approach is the best way to go.

I do want to try and bang out https://github.com/swift-aws/aws-sdk-swift/issues/78 before we build all the repos though. Thoughts?

**Comment by Yasumoto** [comment]
On a relatively beefy EC2 instance:

```
3059.32user 92.78system 4:02.69elapsed 1298%CPU (0avgtext+0avgdata 1691004maxresident)k
```

vs [this branch](https://github.com/swift-aws/aws-sdk-swift/pull/81):

```
1696.18user 62.52system 3:15.81elapsed 898%CPU (0avgtext+0avgdata 1660268maxresident)k
```

I'm curious at the CPU difference, so I'm just going to be bold and make a PR to see what happens to Travis.

**Comment by jonnymacs** [comment]
@noppoMan @Yasumoto fyi I'll be working on #78 this week to unblock this

**Comment by Yasumoto** [comment]
I spent a bit more time looking at #81, and I actually like it a good bit now.

I realized this it's similar to [NIO's approach](https://github.com/apple/swift-nio/blob/master/Package.swift). Even though for _building_ the entire git repository gets pulled along, it doesn't get included in the final product.

I think this makes it much easier to manage/maintain the code, but also simplifies the way to allow folks to include new components in their tools/services. I'm writing something which I thought would only need `Iam`, but now realize I also need ELB, ALB, and CloudFront module as well.

Overall, I'm in favor of keeping maintenance simple, and I think this does that without introducing too much overhead for users.
